compiler/src/dotty/tools/dotc/transform/init/Semantic.scala

```diff
@@ -629,7 +629,9 @@ object Semantic {
               // normal method call
               if argErrors.nonEmpty && isSyntheticApply(meth) then
                 val klass = meth.owner.companionClass.asClass
-                instantiate(klass, klass.primaryConstructor, args, source)
+                val outerCls = klass.owner.lexicallyEnclosingClass.asClass
+                val outer = resolveOuterSelect(outerCls, ref, 1, source)
+                Semantic.instantiate(outer)(klass, klass.primaryConstructor, args, source)
```

Comment:
Why not `outer.instantiate(...)`?

Reply:
Thanks for the catch, this is now updated.